### PR TITLE
Updated parsing logic to keep schema intact

### PR DESCRIPTION
**Pull request by andrewpai**

### CODE REVIEW

**Comment by robotdan** [comment]
Can you update your `.gitignore` to add `.savant/cache` and then revert all of the changes to `.savant` ?

File: build.savant [hunk 1]
```diff
@@ -18,7 +18,7 @@ jacksonVersion = "2.13.4"
 jacksonDatabindVersion = "2.13.4+2"
 testngVersion = "7.3.0"
 
-project(group: "io.fusionauth", name: "fusionauth-scim", version: "2.2.0", licenses: ["ApacheV2_0"]) {
+project(group: "io.fusionauth", name: "fusionauth-scim", version: "2.2.1", licenses: ["ApacheV2_0"]) {
```
Comment (robotdan) [inline build.savant:21]: While in here, let's update the savant plugins to `2.0.0-RC.6` and the `idea` one can be `2.0.0-RC.7`.

Comment (robotdan) [reply build.savant:21]: We can upgrade jackson to `>= 2.15.3`

File: src/test/json/json-patch/add_something_test1.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[ {
```
Comment (robotdan) [inline src/test/json/json-patch/add_something_test1.json]: Based upon your path `SCIMPathPattern` are there any other examples we can add to this to test the new capability - or is this adequate?

Comment (robotdan) [inline src/test/json/json-patch/add_something_test1.json]: Let's rename this test file more appropriately. Ideally mentioning where we saw it - per the naming conventions of the other files. 

Maybe `add_members_nested_schema_azure2.json` -or optionally add to the existing `add_members_azure1` test.

Comment (andrewpai) [reply src/test/json/json-patch/add_something_test1.json]: Added a test case for an enterprise property with a subattribute.

**Review by robotdan — APPROVED** [review]

**Pull request merged**